platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/signed/DefaultSignedStateHasher.java

```diff
@@ -70,23 +70,27 @@ public DefaultSignedStateHasher(
      * {@inheritDoc}
      */
     @Override
-    public void hashState(@NonNull final StateAndRound stateAndRound) {
+    @Nullable
+    public StateAndRound hashState(@NonNull final StateAndRound stateAndRound) {
         final Instant start = Instant.now();
-        try {
+        try (final ReservedSignedState reservedSignedState = stateAndRound.reservedSignedState()) {
             MerkleCryptoFactory.getInstance()
-                    .digestTreeAsync(stateAndRound.reservedSignedState().get().getState())
+                    .digestTreeAsync(reservedSignedState.get().getState())
                     .get();
 
             if (signedStateMetrics != null) {
                 signedStateMetrics
                         .getSignedStateHashingTimeMetric()
                         .update(Duration.between(start, Instant.now()).toMillis());
             }
+
+            return stateAndRound.makeAdditionalReservation("state hasher output");
```

Comment:
this is not the paradigm we should be using, a state reserver should take care of this

Reply:
Pushed a change to just return the `stateAndRound`, rather than obtaining a new reservation, and closing the received one.

The idea was just to more explicitly state which components are using reservations (including transformers), but I see your point.